raiden/utils/signing.py

```diff
@@ -0,0 +1,94 @@
+from coincurve import PrivateKey, PublicKey
+from eth_utils import decode_hex, keccak, remove_0x_prefix, to_bytes, to_checksum_address
+from web3.utils.abi import map_abi_data
+from web3.utils.encoding import hex_encode_abi_type
+from web3.utils.normalizers import abi_address_to_hex
+
+from raiden.exceptions import InvalidSignature
+from raiden.utils.typing import Address, Callable, Optional, Union
+
+sha3 = keccak
+Hasher = Optional[Callable[[bytes], bytes]]
+
+
+def eth_sign_sha3(data: bytes) -> bytes:
+    """
+    eth_sign/recover compatible hasher
+    Prefixes data with "\x19Ethereum Signed Message:\n<len(data)>"
+    """
+    prefix = b'\x19Ethereum Signed Message:\n'
+    if not data.startswith(prefix):
+        data = prefix + b'%d%s' % (len(data), data)
+    return sha3(data)
+
+
+def public_key_to_address(public_key: Union[PublicKey, bytes]) -> Address:
+    """ Converts a public key to an Ethereum address. """
+    if isinstance(public_key, PublicKey):
+        public_key = public_key.format(compressed=False)
+    assert isinstance(public_key, bytes)
+    return to_checksum_address(sha3(public_key[1:])[-20:])
```

Comment:
raiden-libs's `Address` was checksummed, 0x-prefixed `str`, raiden is `bytes`. Need to adapt to `bytes` in this module (returning canonical address instead of checksummed one), as well as fixing code that uses `eth_recover` to prevent it trying to convert from `str`.

Reply:
good point. It works now but types are incorrect. All the places it's called in the client use `to_canonical_address` so might as well make it return a canonical address.

Reply:
should be done